File: app/src/main/java/org/carstenf/wordfinder/Result.java
/*
 * Copyright Carsten Friedrich (Carsten.Friedrich@gmail.com)
 *
 * License: GNU GENERAL PUBLIC LICENSE 3.0 (https://www.gnu.org/copyleft/gpl.html)
 *
 */
package org.carstenf.wordfinder;


import androidx.annotation.NonNull;

/**
 * @author carsten.friedrich@gmail.com
 *
 */
class Result {

	@NonNull
	public String getResult() {
		return result;
	}

	@NonNull
    final private String result;

	Result(	@NonNull String result) {
		this.result=result;
	}

	@NonNull
	@Override
	public String toString() {
		return result;
	}
}

File: app/src/main/kotlin/org/carstenf/wordfinder/util/Result.kt
/*
 * Copyright Carsten Friedrich (Carsten.Friedrich@gmail.com)
 *
 * License: GNU GENERAL PUBLIC LICENSE 3.0 (https://www.gnu.org/copyleft/gpl.html)
 *
 */
package org.carstenf.wordfinder.util

import org.carstenf.wordfinder.dictionary.Dictionary


/**
 * @author carsten.friedrich@gmail.com
 */
class Result(val result: Dictionary.WordInfoData) {
    var isHighlighted: Boolean = false

    override fun toString(): String {
        return result.displayText
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Result

        return result == other.result
    }

    override fun hashCode(): Int {
        return result.hashCode()
    }
}
